app/Encryption/RsaEncryption.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace GatherUp\Encryption;
+
+use phpseclib\Crypt\RSA as CryptRsa;
+
+/**
+ * Uses phpseclib RSA encryption algorithm for generating public/private
+ * keys, encrypting and decrypting messages
+ *
+ * @implements AsymmetricEncryption
+ */
+class RsaEncryption implements AsymmetricEncryption
+{
+    /**
+     * @type CryptRsa
+     */
+    private $rsa;
+
+    /**
+     * Create a RsaEncryption object for creating and using
+     * public and private keys
+     *
+     * @constructor
+     */
+    public function __construct()
+    {
+        $this->rsa = new CryptRsa();
+    }
+
+    /**
+     * Generate an object with a publicKey and privateKey
+     *
+     * @override
+     * @return Object
+     */
+    public function createKeyPair()
+    {
+        $keys = $this->rsa->createKey();
```

Comment:
I think this defaults to a 1028 bit key https://github.com/phpseclib/phpseclib/blob/master/phpseclib/Crypt/RSA.php#L418